### DISCUSSION TITLE
Step vs WorkflowStep

### DISCUSSION THREAD

**Discussion by azro352** [argoproj-labs/hera#564] [Q&A] [answered]
Hi
I just discovered the lib, congrats, it seems nice, I'll try it soon 

Just went on https://hera.readthedocs.io/en/stable/examples/workflows/steps_types/ and I can't figure out the difference between Step and WorkflowStep, they produce the same in the resulting yaml 
Is it just that the first is a custom Hera class and the second an auto generated from Argo specifications ?
Thanks

**Comment by flaviuvadan** [comment, marked as answer]
> I'll try it soon

Awesome! Would love to hear about your use case :) 

> Is it just that the first is a custom Hera class and the second an auto generated from Argo specifications ?

That's right! Hera classes extend the base ones from Argo Workflows and try to simplify things to increase accessibility and QOL in working with Argo. Step is such a class, Task is another, etc

**Reply by azro352** [reply]
Ok, i went to see the full source code, not an extend but kind of a builder with `_build_as_workflow_step`

I'll soon try a complex workflow with some recursion to do files merging 2by2 until there is one left, i'll experiment Hera at this occasion 

Thanks
